src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -461,23 +451,10 @@ public long getVulnerabilityCount(Project project, boolean includeSuppressed) {
      * @param project the Project to retrieve vulnerabilities of
      * @return a List of Vulnerability objects
      */
+    @RegisterBeanMapper(Vulnerability.class)
     public List<Vulnerability> getVulnerabilities(Project project, boolean includeSuppressed) {
-        final List<Vulnerability> vulnerabilities = new ArrayList<>();
-        final List<Component> components = getAllComponents(project);
-        for (final Component component: components) {
-            final Collection<Vulnerability> componentVulns = pm.detachCopyAll(
-                    getAllVulnerabilities(component, includeSuppressed)
-            );
-            Map<String, Epss> matchedEpssList = getEpssForCveIds(
-                    componentVulns.stream().map(vuln -> vuln.getVulnId()).distinct().toList());
-            for (final Vulnerability componentVuln: componentVulns) {
-                componentVuln.setComponents(Collections.singletonList(pm.detachCopy(component)));
```

Comment:
Note how in the original logic, vulnerable components are attached to the vulnerability. And the same vulnerability could appear multiple times in the result, if it affects multiple components.

In the new SQL query, `DISTINCT` is used to de-duplicate based on vulnerability attributes, and no component information is returned at all.